### PR TITLE
Identity service challenge steps

### PR DESCRIPTION
**Pull request by vinit717**
### Issue #663 

parent ticket: https://github.com/Real-Dev-Squad/todo-action-items/issues/178

### What is the change?
Added challenges steps for the developer after filling the join form 

- [x] Yes
- [ ] No

## Example
https://github.com/Real-Dev-Squad/website-www/assets/111434418/ff102eed-9ead-4836-b584-0dc12d847e2c

### CODE REVIEW

File: app/components/identity-steps/step-one.hbs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <p data-test-getting-started-paragraph-1 class="p2-description">Thank you for providing all the details
+        Before joining the community we would want
+        you to complete a small challenge which
+        will also help you in setting up your
+        identity across Real Dev Squad
+    </p>
+    <p data-test-getting-started-paragraph-2 class="p2-description">Please click proceed to know about 
+the task
+    </p>
+    <Reusables::Button @text="Proceed" @variant='dark' @onClick={{@startHandler}} @test='lets-go'
```
Comment (satyam73) [inline app/components/identity-steps/step-one.hbs:12]:
```suggestion
    <Reusables::Button @text="Proceed" @variant='dark' @onClick={{@startHandler}} @test='identity-proceed-button'
```

File: app/components/stepper-signup.hbs [hunk 2]
```diff
@@ -45,18 +45,39 @@
...
-      {{/if}}
-    </div>
-  {{/if}}
+ {{#if (and (gte this.currentStep 3) (lte this.currentStep 6))}}
+  <div class='stepper__buttons'>
+    {{#if (eq this.currentStep 6)}}
+      <Reusables::Button @text='Previous' @variant='light' @onClick={{this.decrementStep}} @test='prev' @type='button' />
+      <Reusables::Button @text='Submit' @variant='dark' @onClick={{this.startHandler}} @test='submit' @type='button' />
+    {{else}}
+      <Reusables::Button data-test-prev-btn @text='Previous' @variant='light' @onClick={{this.decrementStep}} @test='prev' @type='button' />
```
Comment (satyam73) [inline app/components/stepper-signup.hbs:78]: no need to add `data-test-prev-btn` attribute  you're already passing in @test prop

File: app/components/stepper-signup.hbs [hunk 3]
```diff
@@ -45,18 +45,39 @@
...
+      <Reusables::Button data-test-next-btn @text={{if (eq this.currentStep 5) 'Preview' 'Next' }} @variant='dark' @disabled={{if (or this.preValid this.isValid) false true}} @onClick={{this.incrementStep}} @test='next' @type='submit' />
```
Comment (satyam73) [inline app/components/stepper-signup.hbs:79]: similarly here

File: tests/integration/components/identity-steps/step-one-test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'website-www/tests/helpers';
+import { render, click } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | identity-steps/step-one', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
```
Comment (satyam73) [inline tests/integration/components/identity-steps/step-one-test.js:9]: StepOne renders will be good here I think.

File: tests/integration/components/identity-steps/step-one-test.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    this.set('startHandler', async () => {
```
Comment (satyam73) [inline tests/integration/components/identity-steps/step-one-test.js:10]: There's no assertion for this function, how we can be sure that this function worked?

File: tests/integration/components/identity-steps/step-three-test.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import { render, click } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | identity-steps/step-three', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    this.set('startHandler', async () => {
+      await click('[data-test-button=lets-go]');
+    });
```
Comment (satyam73) [inline tests/integration/components/identity-steps/step-three-test.js:12]: there's no assertion for this function

File: tests/integration/components/identity-steps/step-two-test.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import { render, click } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | identity-steps/step-two', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    this.set('startHandler', async () => {
+      await click('[data-test-button=lets-go]');
+    });
```
Comment (satyam73) [inline tests/integration/components/identity-steps/step-two-test.js:12]: no assertions for this function

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: app/components/identity-steps/step-one.hbs (hunk 1 above)
Comment (vinit717) [reply app/components/identity-steps/step-one.hbs]: done

File: app/components/stepper-signup.hbs (hunk 2 above)
Comment (vinit717) [reply app/components/stepper-signup.hbs]: done

File: tests/integration/components/identity-steps/step-one-test.js (hunk 5 above)
Comment (vinit717) [reply tests/integration/components/identity-steps/step-one-test.js]: For now I have remove click handler as this test will cover on stepper-signup test to test the whole integration like after click on button what step should I have to show

File: app/components/identity-steps/step-three.hbs
Comment (satyam73) [inline app/components/identity-steps/step-three.hbs:1]: same as step one, improve class namings and button test prop naming

File: app/components/identity-steps/step-two.hbs
Comment (satyam73) [inline app/components/identity-steps/step-two.hbs:1]: same as step one and three

**Review by satyam73 — CHANGES_REQUESTED** [review]
Tests for the functions can be added

**Review by satyam73 — APPROVED** [review]
LGTM

**Review by shubhamsigdar1 — APPROVED** [review]

**Pull request merged**